src/resource_clients/actor.ts

```diff
@@ -459,3 +462,50 @@ export interface ActorDefinition {
     maxMemoryMbytes?: number;
     usesStandbyMode?: boolean;
 }
+
+type CommonActorPricingInfo = {
+    /** in [0, 1], fraction of pricePerUnitUsd that goes to Apify */
+    apifyMarginPercentage: number;
+    /** when this pricing info record has been created */
+    createdAt: Date;
+    /** since when is this pricing info record effective for a given Actor */
+    startedAt: Date;
+    notifiedAboutFutureChangeAt?: Date;
+    notifiedAboutChangeAt?: Date;
+    reasonForChange?: string;
+}
+
+export type FreeActorPricingInfo = CommonActorPricingInfo & {
+    pricingModel: 'FREE';
+}
+
+export type FlatPricePerMonthActorPricingInfo = CommonActorPricingInfo & {
+    pricingModel: 'FLAT_PRICE_PER_MONTH';
+    /** for how many days this Actor can be used for free in trial period */
+    trialMinutes?: number;
+    /** monthly flat price in USD */
+    pricePerUnitUsd: number;
+}
+
+export type PricePerDatasetItemActorPricingInfo = CommonActorPricingInfo & {
+    pricingModel: 'PRICE_PER_DATASET_ITEM';
+    /** name of the unit that is being charged */
+    unitName?: string;
+    pricePerUnitUsd: number;
+}
+
+export type ActorChargeEvents = Record<string, {
+    eventPriceUsd: number;
+    eventTitle: string;
+    eventDescription?: string;
+}>
+
+export type PricePerEventActorPricingInfo = CommonActorPricingInfo & {
+    pricingModel: 'PAY_PER_EVENT';
+    pricingPerEvent: {
+        actorChargeEvents: ActorChargeEvents
+    };
+    minimalMaxTotalChargeUsd?: number;
```

Comment:
I know the reasoning behind this, since I overheard some Slack conversation, but other ppl might not.

Can we add jsdocs for this (and other properties as well)? 

Also, I know that we won't solve this here, but how about we call it `minimalChargeLimitUsd` (or even `minTotalUsd`)? *Minimal maximum total charge in US dollars* sounds like something straight outta Dr. Seuss book.

Reply:
I forwarded the issue :shrug: